
# HG changeset patch
# User Andrew John Hughes <gnu_andrew@member.fsf.org>
# Date 1396030366 0
# Node ID e8e59c50a451b45bd10cd495b81e5c61b2b4bca8
# Parent  46f9059509fec2498b1a04ccefee538a45aeff8c
PR1726: configure fails looking for ecj.jar before even trying to find javac

2014-03-28  Andrew John Hughes  <gnu.andrew@redhat.com>

	* acinclude.m4:
	(IT_FIND_COMPILER): Check value of ECJ_JAR
	and JAVAC here ...
	(IT_FIND_ECJ_JAR): ... rather than here where
	${JAVAC} may not yet be defined.

diff -r 46f9059509fe -r e8e59c50a451 NEWS
--- a/NEWS	Fri Mar 28 18:06:31 2014 +0000
+++ b/NEWS	Fri Mar 28 18:12:46 2014 +0000
@@ -14,6 +14,9 @@
 
 New in release 2.4.7 (2014-04-XX):
 
+* Bug fixes
+  - PR1726: configure fails looking for ecj.jar before even trying to find javac
+
 New in release 2.4.6 (2014-03-28):
 
 * Backports
diff -r 46f9059509fe -r e8e59c50a451 acinclude.m4
--- a/acinclude.m4	Fri Mar 28 18:06:31 2014 +0000
+++ b/acinclude.m4	Fri Mar 28 18:12:46 2014 +0000
@@ -134,6 +134,12 @@
   IT_USING_ECJ
   IT_WITH_GCJ
   
+  if test "x${ECJ_JAR}" = "xno"; then
+    if test "x${JAVAC}" = "x"; then
+      AC_MSG_ERROR("No compiler or ecj JAR file was found.")
+    fi
+  fi
+
   if test x"${GCJ}" != xno ; then
     JAVAC="${JAVA} -classpath ${ECJ_JAR} org.eclipse.jdt.internal.compiler.batch.Main"
   fi
@@ -350,11 +356,6 @@
       fi
   fi
   AC_MSG_RESULT(${ECJ_JAR})
-  if test "x${ECJ_JAR}" = "xno"; then
-    if test "x${JAVAC}" = "x"; then
-      AC_MSG_ERROR("No compiler or ecj JAR file was found.")
-    fi
-  fi
   AC_SUBST(ECJ_JAR)
 ])
 

